drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -23,6 +23,33 @@ LOG_MODULE_REGISTER(siwx917_wifi);
 
 NET_BUF_POOL_FIXED_DEFINE(siwx917_tx_pool, 1, _NET_ETH_MAX_FRAME_SIZE, 0, NULL);
 
+int get_sl_wifi_rate_mbps(sl_wifi_rate_t mask)
+{
+	const int wifi_rate_table[SL_WIFI_RATE_11AX_MAX + 1][2] = {
+		{SL_WIFI_AUTO_RATE, 0},       {SL_WIFI_RATE_11B_1, 1},
+		{SL_WIFI_RATE_11B_2, 2},      {SL_WIFI_RATE_11B_5_5, 5},
+		{SL_WIFI_RATE_11B_11, 11},    {SL_WIFI_RATE_11G_6, 6},
+		{SL_WIFI_RATE_11G_9, 9},      {SL_WIFI_RATE_11G_12, 12},
+		{SL_WIFI_RATE_11G_18, 18},    {SL_WIFI_RATE_11G_24, 24},
+		{SL_WIFI_RATE_11G_36, 36},    {SL_WIFI_RATE_11G_48, 48},
+		{SL_WIFI_RATE_11G_54, 54},    {SL_WIFI_RATE_11N_MCS0, 7},
+		{SL_WIFI_RATE_11N_MCS1, 14},  {SL_WIFI_RATE_11N_MCS2, 21},
+		{SL_WIFI_RATE_11N_MCS3, 28},  {SL_WIFI_RATE_11N_MCS4, 43},
+		{SL_WIFI_RATE_11N_MCS5, 57},  {SL_WIFI_RATE_11N_MCS6, 65},
+		{SL_WIFI_RATE_11N_MCS7, 72},  {SL_WIFI_RATE_11AX_MCS0, 8},
+		{SL_WIFI_RATE_11AX_MCS1, 17}, {SL_WIFI_RATE_11AX_MCS2, 25},
+		{SL_WIFI_RATE_11AX_MCS3, 34}, {SL_WIFI_RATE_11AX_MCS4, 51},
+		{SL_WIFI_RATE_11AX_MCS5, 68}, {SL_WIFI_RATE_11AX_MCS6, 77},
+		{SL_WIFI_RATE_11AX_MCS7, 86}};
+
+	for (int i = 0; i < SL_WIFI_RATE_11AX_MAX + 1; i++) {
+		if (wifi_rate_table[i][0] == mask) {
+			return wifi_rate_table[i][1];
```

Comment:
Not a big deal, but I would rather declare member rather than an array of two `int`:
```
		if (wifi_rate_table[i].value_hw == mask) {
			return wifi_rate_table[i].value_zephyr;
		}
```